conn.go

```diff
@@ -263,15 +285,18 @@ func (s *Swarm) setupConn(netConn tpt.Conn, isServer bool) (*Conn, error) {
 	defer s.connLock.Unlock()
 
 	// check for it again as it may have been added already. (TOCTTOU)
-	for c := range s.conns {
-		if c.netConn == netConn {
-			return c, nil
-		}
+	if c, ok := s.connByNet[netConn]; ok {
+		return c, nil
```

Comment:
In this case, do we need to close the stream created in line 273?

Reply:
Ah. No, actually, in this case we're kind of screwed. If we close the stream muxer, we close the connection. If we leave it open, the connection is in a broken state.

Honestly, we should get rid of these checks entirely. The user should not be adding net connections to the swarm multiple times. If they do, that's a bug in *their* code.

Reply:
I think these are no longer necessary, it was really more of a safety measure IIRC

Reply:
So, I've gone ahead and removed this check entirely in https://github.com/libp2p/go-peerstream/pull/13/commits/0d3d388651b5efde19aa31bf5ea028dfb61e9c8a. Sound reasonable?